### ISSUE TITLE
Support for task files in Google Drive and similar mounted volumes

### ISSUE THREAD

**Issue by ningsuhen** [GothenburgBitFactory/timewarrior#465]
In MacOS, taskwarrior fails if the task data is located in a Google Drive folder. This seems to have been caused by a feature which is not supported by Google Drive - F_FULLSYNC in https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/fcntl.2.html

(Used in https://github.com/GothenburgBitFactory/libshared/blob/8baf2dbcad9ea24bb9f00a20d912df5da9b45113/src/FS.cpp#L507 )
. Commenting out the code seems to make it work.

Although this can lead to race conditions, using taskwarrior with Google Drive for single person (maybe 2 machines) might not lead to a lot of issues. Can we make this configurable in some way or handle gracefully with warnings?

**Comment by ningsuhen** [comment]
Opened in timewarrior by mistake. moved to https://github.com/GothenburgBitFactory/taskwarrior/issues/2676

**Comment by sruffell** [comment]
I think this should be reopened.

I just installed a new instance of timewarrior and put my database on a Google Drive mount and ran into this issue.  My proposed fix is https://github.com/GothenburgBitFactory/libshared/pull/70. Once that (or something like it) lands, I can submit another pull request to update the submodule in timewarriors code-base.
